radicle/src/storage/git.rs

```diff
@@ -413,6 +415,16 @@ impl Repository {
 
         Ok(refs)
     }
+
+    pub fn resolve_oid(&self, oid: cob::ObjectId) -> Result<cob::ObjectId, git2::Error> {
+        let object_id = oid.normalized_str();
+        if object_id.len() == 40 {
+            Ok(ObjectId::from_str(&object_id).expect("valid object 1 after normalizzation"))
+        } else {
+            let object = self.backend.revparse_single(&object_id)?;
+            Ok(ObjectId::from(object.id()))
+        }
```

Comment:
This is curious; does `normalize_str` change the value such that re-parsing it into an `ObjectId` change the value?

Reply:
No you are right, there is no reason to call again from_str if the value is the correct one

Reply:
Ok, in that case I wonder whether we need `normalize_str` at all? Is it possible that we could use eg. https://doc.rust-lang.org/std/primitive.u8.html#method.leading_zeros instead? Or just not need anything like that?

Reply:
Mh no wait I was talking about the if block not the else

```
   if object_id.len() == 40 {
            Ok(oid)
```

>This is curious; does normalize_str change the value such that re-parsing it into an ObjectId change the value?

When we call `Oid::from_str("76878")` the result is in a string like `76878000000000000000...`, and if we pass this string to `revparse_single` the call fails because git is not able to catch that this is a short sha1.

So the normalize_str is just returning the original sha1 given by the user (modulo some sha1 with 0 at the end)

This solution can be improved by giving the ObjectId the possibility to store the size of the sha1 given by the user, this is also trivial because can be done inside the `from_str` and in this case, the normalized string is just 2 lines of code.

But maybe this can be done in a followup PR?

Reply:
ie. `"ffff"` -> [0,0,0,0,0,0,0,0.ff] -> Oid([0,0,0,0,0,0,0,0.ff])`

Parsing the string from hex into the right byte slice is easier than doing the normalization.

Reply:
>Oh interesting. Then how about we use Oid::from_bytes, and parse the string as having the zeros in front?

This is to allow to have a wrong string? like `76878000000000000000...`?